%YAML 1.2
# SPDX-FileCopyrightText: NVIDIA CORPORATION & AFFILIATES
# Copyright (c) 2021-2022 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
# http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
# SPDX-License-Identifier: Apache-2.0
---
dope:
  ros__parameters:
    # Cuboid dimension in cm x,y,z

    dimensions: {
      "cracker": [16.403600692749023,21.343700408935547,7.179999828338623],
      "gelatin": [8.918299674987793, 7.311500072479248, 2.9983000755310059],
      "meat": [10.164673805236816,8.3542995452880859,5.7600898742675781],
      "mustard": [9.6024150848388672,19.130100250244141,5.824894905090332],
      "soup": [6.7659378051757813,10.185500144958496,6.771425724029541],
      "sugar": [9.267730712890625,17.625339508056641,4.5134143829345703],
      "bleach": [10.267730712890625,26.625339508056641,7.5134143829345703],

      # HOPE objects
      "AlphabetSoup" : [ 8.3555002212524414, 7.1121001243591309, 6.6055998802185059 ],
      "Butter" : [ 5.282599925994873, 2.3935999870300293, 10.330100059509277 ],
      "Ketchup" : [ 14.860799789428711, 4.3368000984191895, 6.4513998031616211 ],
      "Pineapple" : [ 5.7623000144958496, 6.95989990234375, 6.567500114440918 ],
      "BBQSauce" : [ 14.832900047302246, 4.3478999137878418, 6.4632000923156738 ],
      "MacaroniAndCheese" : [ 16.625600814819336, 4.0180997848510742, 12.350899696350098 ],
      "Popcorn" : [ 8.4976997375488281, 3.825200080871582, 12.649200439453125 ],
      "Mayo" : [ 14.790200233459473, 4.1030998229980469, 6.4541001319885254 ],
      "Raisins" : [ 12.317500114440918, 3.9751999378204346, 8.5874996185302734 ],
      "Cherries" : [ 5.8038997650146484, 7.0907998085021973, 6.6101999282836914 ],
      "Milk" : [ 19.035800933837891, 7.326200008392334, 7.2154998779296875 ],
      "SaladDressing" : [ 14.744099617004395, 4.3695998191833496, 6.403900146484375 ],
      "ChocolatePudding" : [ 4.947199821472168, 2.9923000335693359, 8.3498001098632812 ],
      "Mushrooms" : [ 3.3322000503540039, 7.079899787902832, 6.5869998931884766 ],
      "Spaghetti" : [ 4.9836997985839844, 2.8492999076843262, 24.988100051879883 ],
      "Cookies" : [ 16.724300384521484, 4.015200138092041, 12.274600028991699 ],
      "Mustard" : [ 16.004999160766602, 4.8573999404907227, 6.5132999420166016 ],
      "TomatoSauce" : [ 8.2847003936767578, 7.0198001861572266, 6.6469998359680176 ],
      "Corn" : [ 5.8038997650146484, 7.0907998085021973, 6.6101999282836914 ],
      "OrangeJuice" : [ 19.248300552368164, 7.2781000137329102, 7.1582999229431152 ],
      "Tuna" : [ 3.2571001052856445, 7.0805997848510742, 6.5837001800537109 ],
      "CreamCheese" : [ 5.3206000328063965, 2.4230999946594238, 10.359000205993652 ],
      "Parmesan" : [ 10.286199569702148, 6.6093001365661621, 7.1117000579833984 ],
      "Yogurt" : [ 5.3677000999450684, 6.7961997985839844, 6.7915000915527344 ],
      "GranolaBars" : [ 12.400600433349609, 3.8738000392913818, 16.53380012512207 ],
      "Peaches" : [ 5.7781000137329102, 7.0961999893188477, 6.5925998687744141 ],
      "GreenBeans" : [ 5.758699893951416, 7.0608000755310059, 6.5732002258300781 ],
      "PeasAndCarrots" : [ 5.8512001037597656, 7.0636000633239746, 6.5918002128601074 ]
    }

    # 9 element camera matrix (using default from Ketchup demo)
    # Taken from: https://github.com/andrewyguo/dope_training/blob/master/inference/config/camera_info.yaml
    camera_matrix: [
      364.16501736,   0.0,          121.36296296,
      0.0,            364.16501736, 121.36296296,
      0.0,            0.0,          1.0
    ]
